libs/dal/src/dal.service.ts

```diff
@@ -1,16 +1,17 @@
 import mongoose, { Connection, ConnectOptions } from 'mongoose';
 
+export const baseConfig: ConnectOptions = {
+  // AUTO_CREATE_INDEXES is deprecated, use MONGO_AUTO_CREATE_INDEXES
+  autoIndex: process.env.AUTO_CREATE_INDEXES === 'true' || process.env.MONGO_AUTO_CREATE_INDEXES === 'true',
+  maxIdleTimeMS: process.env.MONGO_MAX_IDLE_TIME_IN_MS ? Number(process.env.MONGO_MAX_IDLE_TIME_IN_MS) : 1000 * 30,
+  maxPoolSize: process.env.MONGO_MAX_POOL_SIZE ? Number(process.env.MONGO_MAX_POOL_SIZE) : 50,
+  minPoolSize: process.env.MONGO_MIN_POOL_SIZE ? Number(process.env.MONGO_MIN_POOL_SIZE) : 10,
```

Comment:
No need for any defaults here now
```suggestion
  autoIndex: Boolean(process.env.AUTO_CREATE_INDEXES) === true || Boolean(process.env.MONGO_AUTO_CREATE_INDEXES) === true,
  maxIdleTimeMS: Number(process.env.MONGO_MAX_IDLE_TIME_IN_MS),
  maxPoolSize: Number(process.env.MONGO_MAX_POOL_SIZE),
  minPoolSize: Number(process.env.MONGO_MIN_POOL_SIZE),
```

Reply:
We need to add all of these env vars to WS + Webhook `env.validators.ts` to ensure the defaults populate

Reply:
I'd keep it as is for now. The [env validator comes after imports](https://github.com/novuhq/novu/blob/fcc12b04b3eb478586f32cfdb88ddbc46e791d53/apps/worker/src/bootstrap.ts#L19). It's not the first thing that runs in the code so we might have some unwanted side-effects.

The truth is that we need only one place to set defaults but I want to avoid touching the env validator as part of this PR.

Reply:
Sounds good 👍